### PR TITLE
`LinkModel` class improved fields & methods

### PR DESCRIPTION
**Pull request by benwerner01**
## 🌟 What is the purpose of this PR?

This PR makes some small improvements to the `LinkModel` class, including:
- populating an `ownedById` field (which replaces `accountId`)
- using structural querying in the `getOutgoingLinks` method, and unifying 3 related methods into a single one

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1202882587747080/1202980861294710/f) _(internal)_

## 🔍 What does this change?

See commits/prior description.

## 📜 Does this require a change to the docs?

No.

## 🛡 What tests cover this?

- The existing `link.model.test.ts` file has been updated to test the functionality of the unified `getOutgoingLinks` method.

## ❓ How to test this?

Check CI/run integration tests locally.

## 📹 Demo

image

### CODE REVIEW

File: packages/hash/api/src/model/knowledge/entity.model.ts [hunk 1]
```diff
@@ -266,21 +266,33 @@ export default class {
...
+          ? {
+              eq: [
+                { path: ["type", "versionedUri"] },
+                {
+                  literal: params.linkTypeModel.schema.$id,
+                },
+              ],
+            }
+          : [],
+      ].flat(),
```
Comment (thehabbos007) [inline packages/hash/api/src/model/knowledge/entity.model.ts:293]: Would `flat()` work as intended here? Seems that we're only dealing with a 1-dimensional array, so what would happen to the clauses on flattening?

**Review by thehabbos007 — COMMENTED** [review]
Just a comment here

File: packages/hash/api/src/model/knowledge/entity.model.ts (hunk 1 above)
Comment (benwerner01) [reply packages/hash/api/src/model/knowledge/entity.model.ts:293]: Any object (i.e. `{ ... }`) would remain in the array, any array (i.e. `[...]`) in the array would be flattened. This means that an empty array (i.e. `[]`) would be removed from the array entirely, which was the intended behaviour.

I think the integration tests should confirm that this is working as intended, as the method is able to optionally filter by the `linkTypeModel`.

Comment (thehabbos007) [reply packages/hash/api/src/model/knowledge/entity.model.ts:293]: I see, thank you for clarifying! :)

**Review by thehabbos007 — APPROVED** [review]

**Pull request merged**